crmd/te_events.c

```diff
@@ -565,23 +581,33 @@ process_graph_event(xmlNode * event, const char *event_node)
         desc = "arrived late";
         abort_transition(INFINITY, tg_restart, "Inactive graph", event);
 
-    } else if (match_graph_event(action, event, event_node, status, rc, target_rc) < 0) {
-        desc = "unknown";
-        abort_transition(INFINITY, tg_restart, "Unknown event", event);
+    } else {
+        action = get_action(action_num, FALSE);
 
-    } else if (rc == target_rc) {
-        passed = TRUE;
-        crm_trace("Processed update to %s: %s", id, magic);
+        if (action == NULL) {
+            desc = "unknown";
+            abort_transition(INFINITY, tg_restart, "Unknown event", event);
+
+        } else {
+            /* XML_ATTR_TE_ALLOWFAIL will be true if on-fail=ignore for the operation */
+            ignore_failures = crm_is_true(crm_meta_value(action->params,
+                                                         XML_ATTR_TE_ALLOWFAIL));
```

Comment:
That was my initial plan but I needed ignore_failures (from action) for update_failcount() later. I could have done it with new return codes from match_graph_event() but this seemed just as easy.